### ISSUE TITLE
Acknowledge contributors

### ISSUE THREAD

**Issue by vlsi** [javacc/javacc-8-core#1]
javacc-8-core contains code contributed by several people, however its git history is blank: https://github.com/javacc/javacc-8-core/graphs/contributors

For instance, I contributed this code: https://github.com/javacc/javacc-8-core/blob/59a28984c1977444c40d11d71e5d32908500d05d/src/main/java/org/javacc/parser/Expansion.java#L128-L142 , and I want that the change to be attributed appropriately.

**Comment by new-javacc** [comment]
Sure you can continue and get stuck in 7 lol - if you prefer. 

I want @vlsi to be useful and propose fixes rather than just keep saying it can't be done. Or just fork it and take it. But I know what my goal is - the new way of doing core is the ONLY way forward and if you have problem with it - take one of the other options I proposed there. I don't have the time/wherewithal to do the git stuff.

**Comment by vlsi** [comment]
@new-javacc , I don't understand how this issue is connected with "continuing with 7".
I just say that the history of 8 should be recollected so it includes all the contributors (not just me).

**Comment by new-javacc** [comment]
So please go ahead and do it if you know how to! I have no idea how to do it or the patience to figure out. Like @zosrothko  said we can add contributors names etc.

**Comment by vlsi** [comment]
To my best knowledge, it would require combining the repositories. Are you ok with that?

**Comment by new-javacc** [comment]
As long as we have the new way forward in the structure I want (core and codegen separate), I don't see a problem with it I simply do not to support new work (or even bug fixes after some time) in the previous version(s) myself.

**Comment by vlsi** [comment]
> core and codegen separate

Is separate folders enough?

**Comment by new-javacc** [comment]
No I want separate modules or repos. I want people to be able to checkout just code generator they want and work on it

**Comment by new-javacc** [comment]
I want the separation to reflect strongly in the code organization.

**Comment by vlsi** [comment]
>I want people to be able to checkout just code generator they want and work on it

What's wrong if they checkout everything and they start working on the bits they need?

**Comment by new-javacc** [comment]
No - I don't want it that way. I want the core to actually become sometehing like javaccc - javacc compiler! I want a very strong dissociation between the two

**Comment by new-javacc** [comment]
Also this is just your personal obsession @vlsi  (whereas the other one is my personal obsession lol). So what do others think?Maybe I will open a poll

**Comment by zosrothko** [comment]
> What's wrong if they checkout everything and they start working on the bits they need?
That is wrong because the version number HAS A MEANING. It should cortrespond to added/removed/changed part of the code/documentation/etc... We DO NOT WANT a new version of the core for exemple if there is no change in it while a other module like the JAva code generator has been changed. Could you get it?

**Comment by zosrothko** [comment]
I will try those steps described [there](https://stackoverflow.com/questions/1683531/how-to-import-existing-git-repository-into-another). If the result is positive, it will be pushed.

**Comment by new-javacc** [comment]
Thanks, Francis! Others who are so hung up on this can maybe help?On Jul 10, 2022 11:43 PM, Francis ANDRE ***@***.***> wrote:
I will try those steps described there. If the result is positive, it will be pushed.

**Comment by vlsi** [comment]
>Others who are so hung up on this can maybe help?

@new-javacc , I can help with recollecting the history, however, it would result in merging all the repositories into a single repository as I said in https://github.com/javacc/javacc-8-core/issues/1#issuecomment-1179775877

Please let me know once you decide to go with a single repository instead of multiple ones.

**Comment by zosrothko** [comment]
keeping back the javacc-7 history into javacc-8-core is done. Look at javacc-8-core/contributors.

**Comment by new-javacc** [comment]
It has to be multiple mostly independent repos with only code gen depending on core. That's my decision for the future verison of javacc based on what I know of the codebase, customers and contributors.

**Comment by vlsi** [comment]
The git history is still broken though.

Please compare git blame for `Expansion.java` in 7 and 8

7: https://github.com/javacc/javacc/blame/31febfb562b80f6112a3c873aa6097078223659f/src/main/java/org/javacc/parser/Expansion.java#L125-L143

8: https://github.com/javacc/javacc-8-core/blame/1da102ab29ef38f0ad5a1610ac7892170fb80422/src/main/java/org/javacc/parser/Expansion.java#L125-L142

Basically, in javacc-8-core repository, it looks the file has been created by @zosrothko in a single commit.
However, in javacc-7, there's clear history.

**Comment by zosrothko** [comment]
On the insigth contributors of javacc-8-core, all your commits appears to be yours

![image](https://user-images.githubusercontent.com/7262593/178327062-9407624f-95e7-4049-8fdd-9936d14667cc.png)

**Comment by vlsi** [comment]
@zosrothko , please check the URLs I listed:
https://github.com/javacc/javacc-8-core/blame/1da102ab29ef38f0ad5a1610ac7892170fb80422/src/main/java/org/javacc/parser/Expansion.java#L125-L142

<img width="923" alt="git blame for javacc-8" src="https://user-images.githubusercontent.com/213894/178329802-73b9f887-ea8e-40cc-8356-a52ad4943e30.png">

The same link for javacc-7 properly shows which lines have been modified by which commit:
<img width="1053" alt="git blame for javacc-7" src="https://user-images.githubusercontent.com/213894/178329910-a1c194e8-9848-4107-8de2-14f2b5f3b610.png">

**Comment by new-javacc** [comment]
That's probably because we split out 8 before your fix. Also note that it's a codegen fix so not relevant anymore.On Jul 11, 2022 11:11 AM, Vladimir Sitnikov ***@***.***> wrote:
@zosrothko , please check the URLs I listed:
https://github.com/javacc/javacc-8-core/blame/1da102ab29ef38f0ad5a1610ac7892170fb80422/src/main/java/org/javacc/parser/Expansion.java#L125-L142

The same link for javacc-7 properly shows which lines have been modified by which commit:

**Comment by vlsi** [comment]
>we split out 8 before your fix. Also note that it's a codegen fix so not relevant anymore

8 was split after my fix, and the change is still needed even for 8.

**Comment by new-javacc** [comment]
> > we split out 8 before your fix. Also note that it's a codegen fix so not relevant anymore
> 
> 8 was split after my fix, and the change is still needed even for 8.

Did you have a repro and test case? I haven't looked at the fix. Send me the PR that was merged and I can look. The title says good names for jj methods - which is a codegen issue. So not relevant in 8. Not sure what this loop is doing.

**Comment by vlsi** [comment]
Here's the PR https://github.com/javacc/javacc/pull/153

The change is present in the most recent 8 code, so I don't understand what do you mean by "not relevant in 8"

**Comment by new-javacc** [comment]
You are wasting so much of my time it's annoying.That function getProductionName is unused. If it is in the codegen, I will remove it. I want everything related to codegen for a specific language to in that module/repo, not in core. So I want to refactor to get rid of this function in core and move it to codegen - maybe as a Util class or something

**Comment by new-javacc** [comment]
Basically, there is no going back from having core as a separate repo and each language code generator in their own repos. Rest of the stuff history etc whoever is passionate about and have a ton of time on their hands, please do it. And stop calling Francis names. He tried to do his best working with me (not an easy task) with my constraints so I'm really thankful to him.

**Comment by MarcMazas** [comment]
Can't do more that add Vlad to the pom's contributors list.
